tests/foreman/virtwho/test_api_virtwho.py

```diff
@@ -67,6 +71,26 @@ def _make_virtwho_configure(self):
             args[u'hypervisor_password'] = self.hypervisor_password
         return args
 
+    def _try_to_get_guest_bonus(self, hypervisor_name, sku):
+        vdc_id = None
+        subscriptions = entities.Subscription().search(
+            query={'search': '{0}'.format(sku)})
+        for item in subscriptions:
+            item = item.read_json()
+            if hypervisor_name.lower() in item['hypervisor']['name']:
+                vdc_id = item['id']
+                break
+        return vdc_id
+
+    def _get_guest_bonus(self, hypervisor_name, sku):
```

Comment:
Could we combine these two functions(` _try_to_get_guest_bonus` and `_get_guest_bonus`) to one?

Reply:
I tried the `wait_for`, and it works fine, plz see my next commits, thanks

Reply:
And we should filter it through hypervisor_name after we getting the searching result, so I still need 2 functions